TTHAnalysis/python/tools/higgsDiffGenTTH.py

```diff
@@ -227,71 +399,168 @@ def analyze(self, event):
         self.out.fillBranch('%snQFromWFromT'%self.label     , len(QFromWFromT))
         self.out.fillBranch('%snLFromWFromH'%self.label     , len(LFromWFromH))
         self.out.fillBranch('%snLFromWFromT'%self.label     , len(LFromWFromT)) 
+        self.out.fillBranch('%snTauFromH'%self.label        , len(TauFromH))
+        self.out.fillBranch('%snhFromTauFromH'%self.label   , len(hFromTauFromH))
+        self.out.fillBranch('%snLFromTauFromH'%self.label   , len(LFromTauFromH))
+        self.out.fillBranch('%snNuFromTauFromH'%self.label  , len(NuFromTauFromH))
+        self.out.fillBranch('%snTNuFromTauFromH'%self.label , len(TNuFromTauFromH))
+        self.out.fillBranch('%snZFromH'%self.label          , len(ZFromH))
+        self.out.fillBranch('%snQFromZFromH'%self.label     , len(QFromZFromH))
+        self.out.fillBranch('%snLFromZFromH'%self.label     , len(LFromZFromH))
+        self.out.fillBranch('%snNuFromZFromH'%self.label    , len(NuFromZFromH))
+        self.out.fillBranch('%snTauFromWFromH'%self.label   , len(TauFromWFromH))
+        self.out.fillBranch('%snLFromTauFromWFromH'%self.label  , len(LFromTauFromWFromH))
+        self.out.fillBranch('%snNuFromTauFromWFromH'%self.label , len(NuFromTauFromWFromH))
+        self.out.fillBranch('%snhFromTauFromWFromH'%self.label  , len(hFromTauFromWFromH))
+
+        self.out.fillBranch('%sHiggses_pt'%self.label               , [ Higgses[i].p4().Pt()               if len(Higgses) > i             else -99 for i in range(4) ]) 
```

Comment:
Now I see it. This consumes more space in that it fills the arrays up to the maximum dimension. 
The current structure fills jagged arrays, please let's keep it. Can you please harmonize the new variables you added to the current filling structure and re-commit?